### PR TITLE
PID for shooter

### PR DESCRIPTION
**Pull request by hazeltorek**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/OuttakeSubsystem.java [hunk 1]
```diff
@@ -8,57 +8,108 @@
...
+   * Check if the shooter is at its setpoint, within the error margin set by setpointError.
+   * 
+   * @return true if it is within the margin of error
+   */
+  public boolean upToSpeed() {
+    // Units are in encoder units per 100 ms right now
+    double velocity = shootMotor1.getSensorCollection().getIntegratedSensorVelocity();
+
+    // dimensional analysis????????????????
+    double rpm = velocity * 29.296875;
```
Comment (20MatthewNeal) [inline src/main/java/frc/robot/subsystems/OuttakeSubsystem.java]: Magical number

File: src/main/java/frc/robot/subsystems/OuttakeSubsystem.java [hunk 2]
```diff
@@ -8,57 +8,108 @@
...
+    shootMotor2.follow(shootMotor1);
+
+    shootMotor1.config_kP(1, P);
+    shootMotor2.config_kP(1, P);
 
-    shootMotor1 = new WPI_TalonFX(5);
-    shootMotor2 = new WPI_TalonFX(6);
-    feederMotor = new WPI_TalonSRX(7);
+    shootMotor1.config_kD(1, D);
+    shootMotor2.config_kD(1, D);
```
Comment (20MatthewNeal) [inline src/main/java/frc/robot/subsystems/OuttakeSubsystem.java:43]: statically imported constants, comment please.

**Review by 20MatthewNeal — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/OuttakeSubsystem.java [hunk 3]
```diff
@@ -8,57 +8,109 @@
...
+    shootMotor2.selectProfileSlot(1, 0);
   }
 
    /**
     * Sets speed of motors in order to shoot in low goal
     */
   public void shootLow(){
+    // Open loop control is used on feed motors
+    if (upToSpeed()) {
+      // Only feed is the shooter is ready
```
Comment (MatthewEdgmon) [inline src/main/java/frc/robot/subsystems/OuttakeSubsystem.java]: change is to if

File: src/main/java/frc/robot/subsystems/OuttakeSubsystem.java [hunk 4]
```diff
@@ -8,57 +8,109 @@
...
+   * @return the current setpoint of the shooter
+   */
+  public double getSetpoint() {
+    return setpoint;
   }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
+    shootMotor1.set(ControlMode.Velocity, setpoint);
```
Comment (MatthewEdgmon) [inline src/main/java/frc/robot/subsystems/OuttakeSubsystem.java:107]: Do we need shootMotor2 as well?

**Review by MatthewEdgmon — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/OuttakeSubsystem.java (hunk 4 above)
Comment (MatthewEdgmon) [reply src/main/java/frc/robot/subsystems/OuttakeSubsystem.java:107]: Nope.

**Review by MatthewEdgmon — APPROVED** [review]

**Review by 20MatthewNeal — APPROVED** [review]

**Pull request merged**